wrapanapi/virtualcenter.py

```diff
@@ -769,3 +769,57 @@ def usage_and_quota(self):
             'cpu_total': installed_cpu,
             'cpu_limit': None,
         }
+
+    def add_disk_to_vm(self, vm_name, capacityInMB, datastore=None, type=None, unit=None):
+        """
+        Create a disk on the given datastore (by name)
+
+        Community Example used:
+        https://github.com/vmware/pyvmomi-community-samples/blob/master/samples/add_disk_to_vm.py
+
+        :param name: string name of the disk
+        :param capacityInMB: integer capacity, number of MB
+        :param datastore: the name of the datastore
+        :param type: string 'thin' or 'thick'
+        :param unit: unit number of disk, careful not to overwrite existing units
+        :return: boolean disk creation result
+        """
+        type = type if type in ['thick', 'thin'] else 'thin'
+        vm = self._get_vm(vm_name=vm_name)
+
+        if not unit:
+            unit = 0
+            # user didn't pass a disk unit number, select next available
+            for dev in vm.config.hardware.device:
+                if hasattr(dev.backing, 'fileName'):
+                    unit = int(dev.unitNumber) + 1
```

Comment:
this block isn't quite working with our providers, fixing it up to better detect an existing VirtualDisk device